### PR TITLE
v1: builder proposer endpoint

### PR DESCRIPTION
**Pull request by xenowits**
This is a draft of the new `/eth/v1/validator/proposer_config` endpoint.

ticket: https://github.com/ObolNetwork/charon/issues/1643

### CODE REVIEW

**Comment by xenowits** [comment]
Screenshot 2023-03-20 at 18 33 59
Screenshot 2023-03-20 at 18 34 12

**Comment by xenowits** [comment]
How it looks like:
Screenshot 2023-03-27 at 20 46 34
Screenshot 2023-03-27 at 20 40 48

Need review on the text, especially the API summary 🙏

**Review by gsora — APPROVED** [review]
LFTM but let's wait for @corverroos opinion before merging

**Review by corverroos — APPROVED** [review]

File: apis/validator/proposer_config.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+get:
+  tags:
+    - ValidatorRequiredApi
+    - Validator
+  operationId: "proposerConfig"
+  summary: "Produce a proposer config"
+  description: |
+    Requests the beacon node or DVT middleware to produce the proposer config. The proposer_config enables validator registrations
```
Comment (OisinKyne) [inline apis/validator/proposer_config.yaml:8]:
```suggestion
    Requests the beacon node or DV middleware to produce the proposer config. The proposer_config enables validator registrations
```

We should align with Brett on this brand wide, but I prefer DV most of the time 😅

**Review by OisinKyne — APPROVED** [review]
One nit about DVT middleware or DV Middleware that I might not even be right on, but broadly happy to merge to main of our repo, and we work on the support side before we open on the real beacon-api repo.

**Pull request merged**